src/ripple/app/consensus/RCLConsensus.cpp

```diff
@@ -213,13 +213,22 @@ RCLConsensus::Adaptor::propose(RCLCxPeerPos::Proposal const& proposal)
         proposal.prevLedger().begin(), proposal.prevLedger().size());
     prop.set_proposeseq(proposal.proposeSeq());
     prop.set_closetime(proposal.closeTime().time_since_epoch().count());
+
+    if(!validatorKeys_.keys)
+    {
+        JLOG(j_.warn()) << "RCLConsensus::Adaptor::propose: ValidatorKeys "
+                            "not set: \n";
+        return;
+    }
```

Comment:
Since you know that `keys` is set at this point, it might make sense, improve readability a little, and maybe speed things up by an amount too small to measure to set a reference and use it for the rest of the function:
```
auto const& keys = *validatorKeys_.keys;
```